src/rust/ide/src/model/project.rs

```diff
@@ -91,6 +94,12 @@ pub trait API:Debug {
     fn subscribe(&self) -> Subscriber<Notification>;
 }
 
+// Note: Needless lifetimes
+// ~~~~~~~~~~~~~~~~~~~~~~~~
+// Clippy complains about using explicit lifetimes, however the suggested change breaks compilation.
+// Apparently `BoxFuture` must be given an explicit lifetime, and this lifetime cannot be anonymous.
+// Thus, we silent the warning manually.
+
```

Comment:
Could you please report it to the clippy issue tracker and link it here?

Reply:
I've been unable to provide self-contained repro.
Clippy is correct, it seems that `automock`-generated code causes the issue.
The comment has been updated.